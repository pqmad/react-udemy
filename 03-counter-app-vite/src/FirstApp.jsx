/*import { Fragment } from "react"
//rafc
export const FirstApp = () => {
  return (
    <Fragment>
      <h1>FirstApp</h1>
      <p>Ariana</p>
    </Fragment>

  )
}
*/

//const nombre='Arianaaaa';
import PropTypes from "prop-types";


const nombre={
  n1:"ariana",
  edad:22
};
const getsaludo = () => {return "Holis"};


export const FirstApp = ({titulo, subtitulo}) => {
  //console.log(props);
 
  return (
    <>
      <h1>FirstApp</h1>
      <h1 data-testid="test-titulo">{titulo}</h1>
      <h3>{subtitulo}</h3>
      <h3>{subtitulo}</h3>
      <h2>{getsaludo()}</h2>
      <h3>{nombre.edad}</h3>
      <h3>{JSON.stringify(nombre)}</h3>
    </>

  )
}

FirstApp.propTypes = {
  titulo: PropTypes.string.isRequired,
  subtitulo: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
  //titulo: 'No hay titulo',
  subtitulo: 0,
  name:'Madeline'
}