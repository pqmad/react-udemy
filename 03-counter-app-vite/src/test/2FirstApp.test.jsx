import { render,screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp";


describe('pruebas FIRSTAPP', () => { 
  const titulo="holis soy Madeline"; 
  const subtitulo=34;
  
        
    test('debe de hacer match con el snapshot', () => {
      const {container}=render (<FirstApp titulo={titulo} subtitulo={subtitulo}/>)
      expect(container).toMatchSnapshot();
     });

    test('mostrar el titulo en h1', () => { 
      render (<FirstApp titulo={titulo} subtitulo={subtitulo}/>)
      expect(screen.getByText(titulo)).toBeTruthy();
      expect(screen.getByTestId("test-titulo").innerHTML).toContain(titulo);
    });

    test('mostrar el subtitulo', () => {
        render (<FirstApp titulo={titulo} subtitulo={subtitulo}/>)
        expect(screen.getAllByText(subtitulo).length).toBe(2);
        
    });
 })