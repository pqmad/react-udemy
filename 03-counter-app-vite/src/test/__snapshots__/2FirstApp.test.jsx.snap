// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas FIRSTAPP debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    FirstApp
  </h1>
  <h1
    data-testid="test-titulo"
  >
    holis soy Madeline
  </h1>
  <h3>
    34
  </h3>
  <h3>
    34
  </h3>
  <h2>
    Holis
  </h2>
  <h3>
    22
  </h3>
  <h3>
    {"n1":"ariana","edad":22}
  </h3>
</div>
`;
